src/coffeescript.coffee

```diff
@@ -271,111 +249,145 @@ parser.yy.parseError = (message, {token}) ->
   # from the lexer.
   helpers.throwSyntaxError "unexpected #{errorText}", errorLoc
 
-# Based on http://v8.googlecode.com/svn/branches/bleeding_edge/src/messages.js
-# Modified to handle sourceMap
-formatSourcePosition = (frame, getSourceMapping) ->
-  filename = undefined
-  fileLocation = ''
-
-  if frame.isNative()
-    fileLocation = "native"
-  else
-    if frame.isEval()
-      filename = frame.getScriptNameOrSourceURL()
-      fileLocation = "#{frame.getEvalOrigin()}, " unless filename
+# Use native source maps rather than monkey patching `Error.prepareStackTrace`
+# if native source maps are enabled. Do not patch `Error.prepareStackTrace`
+# if another library has patched it since that would break them.
+nativeSourceMaps = process?.execArgv.includes('--enable-source-maps') or process?.env.NODE_OPTIONS?.includes('--enable-source-maps')
```

Comment:
Shouldn’t you define this variable above its first reference? I see it read on line 64 (inside the `compile` function), but defined on line 255. Assuming there are no calls to `compile` before line 255, this will work, since `nativeSourceMaps` will get defined before it’s read; but this feels brittle, because any calls to `compile` above line 255 will cause this to break (I think?).

Anyway this is a one-liner, perhaps we should just move it and its comment to near the top of the file?

Reply:
👍